subsys/bluetooth/host/hci_core.c

```diff
@@ -2936,6 +2958,30 @@ static void hci_encrypt_key_refresh_complete(struct net_buf *buf)
 }
 #endif /* CONFIG_BT_SMP || CONFIG_BT_BREDR */
 
+#if defined(CONFIG_BT_TESTING)
+static void bt_hci_evt_read_remote_version_complete(struct net_buf *buf)
+{
+	struct bt_hci_evt_remote_version_info *evt;
+	evt = net_buf_pull_mem(buf, sizeof(*evt));
+
+	u16_t handle = sys_le16_to_cpu(evt->handle);
+	struct bt_conn *conn;
+
+	conn = bt_conn_lookup_handle(handle);
+	if (!conn) {
+		BT_ERR("Unable to lookup conn for handle %u", handle);
+		return;
+	}
+
+	if (!evt->status) {
+		conn->rv.version = evt->version;
+		conn->rv.manufacturer = evt->manufacturer;
```

Comment:
I am pretty sure these fields need to be endianness-adjusted (cf. the conversion of evt->handle)

Reply:
I don't think so seeing as the example data I have received is only valid un-adjusted. My personal phone's bluetooth chip manufacturer 0x000f is a valid company identifier (Broadcom Corporation), while 0x0f00 is not valid. Highest company identifier is as of today "2012   0x07DC 'ThingOS GmbH'". https://www.bluetooth.com/specifications/assigned-numbers/company-identifiers/

Reply:
I assume you run on a little-endian architecture? sys_le16_to_cpu() is a no-op on a little-endian architecture, but necessary on a big-endian architecture to read the correct 0x000f instead of 0x0f00.

Reply:
I think it's running in little-endian architecture, yes.
So you want this:?
´if (!evt->status) {
		conn->rv.version = sys_le16_to_cpu(evt->version);
		conn->rv.manufacturer = sys_le16_to_cpu(evt->manufacturer);
		conn->rv.subversion = sys_le16_to_cpu(evt->subversion);
	} ´

Reply:
Yes, I believe that would be the correct version.

Reply:
Pushed the change now